src/shared.c

```diff
@@ -1026,6 +1026,14 @@ _Bool ValidateUserInfo(const char *s) {
 void SetUserInfo(char *s, const char *key, const char *value) {
 	char newi[MAX_USER_INFO_STRING], *v;
 
+	if (!value) {
+		return;
+	}
+
+	if (*value == '\0') {
+		return;
+	}
+
```

Comment:
There's no reason to separate them. All compilers short-circuit compound conditionals.